### ISSUE TITLE
Which RDBMS support you want to see in linq2db?

### ISSUE THREAD

**Issue by MaceWindu** [linq2db/linq2db#1014]
Please, don't start any discussion in this thread. Only add new RDBMS or vote for existing one.
Important notes:
- we cannot add support for database if it doesn't have .net provider. You still can add it here, but we will not start working on it without provider
- we cannot add support for database if it is not available for testing. Still it is a good candidate for 3rd-party linq2db extension, like [DB2 iSeries](https://github.com/LinqToDB4iSeries/Linq2DB4iSeries)
- check list of [already supported databases](https://github.com/linq2db/linq2db/wiki#supported-databases) before posting new request

**Comment by MaceWindu** [comment]
[Amazon Redshift](https://en.wikipedia.org/wiki/Amazon_Redshift)
Tracked by https://github.com/linq2db/linq2db/issues/886

**Comment by MaceWindu** [comment]
[Oracle using dotConnect for Oracle provider](https://www.devart.com/dotconnect/oracle/)
Tracked by https://github.com/linq2db/linq2db/issues/946

**Comment by MaceWindu** [comment]
MS Access using ODBC provider
Tracked by https://github.com/linq2db/linq2db/issues/333

**Comment by MaceWindu** [comment]
[Cassandra using DataStax ADO.NET provider](https://docs.datastax.com/en/developer/csharp-driver/3.0/features/components/adonet/)
Tracked by https://github.com/linq2db/linq2db/issues/552

**Comment by MaceWindu** [comment]
Sybase using ODBC
Tracked by https://github.com/linq2db/linq2db/issues/499

**Comment by MaceWindu** [comment]
Sybase SQL Anywhere
Tracked by https://github.com/linq2db/linq2db/issues/186

**Comment by MaceWindu** [comment]
DataAction AdoNetCore.AseClient

**Comment by glenn-slayden** [comment]
Target a set of simple CSV files, 1 per RDBMS table, accessible in a (given) writeable file system directory (table name == filename), where the directory also contains one additional special text-file containing the DB schema (column names plus strong types for all tables) in a simple line-oriented or CSV text format.

See http://moin.delph-in.net/TsdbTop,  for example

**Comment by sdanyliv** [comment]
Connect to MySql via [MySqlConnector](https://github.com/mysql-net/MySqlConnector)

**Comment by tb-mtg** [comment]
The link below can help you setup a free testing environment for [DB2 iSeries](https://github.com/LinqToDB4iSeries/Linq2DB4iSeries):

https://www.nicklitten.com/free-ibm-i-system-for-developers

**Comment by AlexandrSitdikov** [comment]
[Yandex ClickHouse](https://clickhouse.yandex/)

**Comment by sdanyliv** [comment]
New Microsoft’s SqlClient when it become non-preview.
https://github.com/dotnet/SqlClient

**Comment by MaceWindu** [comment]
SAP HANA .net core provider support https://stackoverflow.com/a/56167626/1641529

**Comment by JTrotta** [comment]
RQLITE.
[https://github.com/rqlite/rqlite](https://github.com/rqlite/rqlite)

**Comment by sdanyliv** [comment]
@JTrotta, unfortunately RQLITE does not have .NET support.

**Comment by JTrotta** [comment]
@sdanyliv could [this](https://github.com/zerouid/RQLite.Net) be a good start?

**Comment by MaceWindu** [comment]
@JTrotta , I'm afraid, no. linq2db works with providers over [ADO.NET interfaces](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/data-providers#core-objects-of-net-framework-data-providers) and I don't see implementation for them in referenced code. We cannot afford implementing it on our side.

**Comment by sdanyliv** [comment]
@JTrotta, It should support ADO.NET interfaces `IDbConnection`, `IDbCommand`, etc. Otherwise it is out of scope for current project.

**Comment by MaceWindu** [comment]
@JTrotta , I'm not familiar with RQLITE, but maybe it is possible to work with it using SQLite providers? Or it is too different from SQLite?

**Comment by JTrotta** [comment]
@MaceWindu It uses, as its backend store, System.Data.SQLite.Core, But it uses http commands, just like Elasticsearch, because its distributed on clustered nodes. Its very powerful and fast.
I think it's possible to write a driver starting from [here](https://github.com/linq2db/linq2db/tree/master/Source/LinqToDB/DataProvider/SQLite)

**Comment by sdanyliv** [comment]
@JTrotta, it will be complicated without ADO.NET Interfaces support. When someone write `IDbConnection`, `IDbCommand`, etc. implementation, it will be easy.

For distributed RDBMS i'm looking at [TIDB](https://github.com/pingcap/tidb), which is more mature.

Please note that we will clear this thread to do not overweight original topic. If you want to continue conversation - create new issue.

**Comment by thyn** [comment]
Any news about clickhouse?

**Comment by MaceWindu** [comment]
@thyn let's move discussion to #1796

**Comment by filippobottega** [comment]
[OSQuery](https://osquery.io/)

**Comment by MaceWindu** [comment]
@filippobottega, don't see how we can support it. We can support only databases, available through ADO.NET api. OSQuery doesn't have ado.net provider, so only way to support it - is to use OleDb/ODBC (preferred) provider, but briefly looking - I cannot find any.

**Comment by filippobottega** [comment]
@MaceWindu, you are right, I have tried to search for an ODBC driver without luck like you.
I would like to mention OSQuery because it's really interesting and I hope someone will be able to create a binding between OSQuery and .NET in some way...

**Comment by MaceWindu** [comment]
SAP/Sybase IQ #2040

**Comment by gamblen** [comment]
[Ingres](https://esd.actian.com/product/drivers/.Net_Data_Provider/Windows_64-Bit/.Net_Data_Provider_GA)

**Comment by ComaCluster9981** [comment]
Snowflake https://www.snowflake.com/ , Cloud Data Platform (and DWH)
Provides a .Net driver
https://github.com/snowflakedb/snowflake-connector-net

**Comment by tb-mtg** [comment]
MS Access databases (.accdb) using ODBC

**Comment by tb-mtg** [comment]
IBM iSeries databases using ODBC

**Comment by MaceWindu** [comment]
@tb-mtg, iSeries driver maintained by [3rd party project](https://github.com/LinqToDB4iSeries/Linq2DB4iSeries), so better luck to ask for ODBC support there.

**Comment by sdanyliv** [comment]
CockroachDB, looks like it is Postgres compatible.
https://www.cockroachlabs.com/docs/stable/build-a-csharp-app-with-cockroachdb.html

**Comment by caliskanersin** [comment]
Clickhouse https://clickhouse.tech/

**Comment by sdanyliv** [comment]
@caliskanersin, duplicate https://github.com/linq2db/linq2db/issues/1014#issuecomment-463577679
Will remove this duplicate message.

**Comment by Int32Overflow** [comment]
PostgreSQL with Npgsql driver

**Comment by sdanyliv** [comment]
> PostgreSQL with Npgsql driver

What does it mean? linq2db supports PostgreSQL via Npgsql driver.

**Comment by travbeamo** [comment]
Support for DB2iSeries using Odbc and/or OleDb: 
See https://github.com/linq2db/linq2db/issues/2473 for details.

**Comment by gjvanvuuren** [comment]
An Azure Synapse provider for Microsoft Sql Azure

**Comment by filippobottega** [comment]
Support for [OpenSQL](https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-US/abenopensql.htm)
In SAP the access to the database is done using OpenSQL language, an extension of ABAP language used to access the db.
The idea could be to use linq2db to emit the OpenSQL embedded in a Function Module and let the user to create the emitted function module in a SAP package. Then linq2db could execute the query calling the function module through the [SAP Connector for Microsoft .NET 3.0 ](https://support.sap.com/en/product/connectors/msnet.html). Linq2db could use the connector to read the table definitions from the ABAP Dictionary allowing the user to use them in the definition of queries written in .NET code.
Any better idea to extract data from SAP using linq2db?

**Comment by yallie** [comment]
PostgreSQL using a free edition of [dotConnect for PostgreSQL data provider](https://www.devart.com/dotconnect/postgresql/) (for .NET 4.x support).

**Comment by sdanyliv** [comment]
@yallie, which benefits it gives over Npgsql?

**Comment by yallie** [comment]
@sdanyliv,

1. No dependencies, see [Nuget](https://www.nuget.org/packages/dotConnect.Express.for.PostgreSQL/)
2. Npgsql 5.x has dropped support for the older versions of .NET framework

**Comment by nomada2** [comment]
any plans for Azure Cosmos DB?

**Comment by sdanyliv** [comment]
@nomada2,  I don't think that Cosmos DB is relative to this library. It is not RDBMS but more like NoSQL database with some SQL support.

**Comment by dagtveit** [comment]
Clickhouse. which is a timeseries database but i would say its more relational then most others. and already supports postgress language ? any chance @MaceWindu. also would like to discuss potential support for doing it if we are not talking crazy numbers ?

**Comment by mguinness** [comment]
While not strictly a RDBMS, Windows Search service allows for [Querying the Index with Windows Search SQL Syntax](https://docs.microsoft.com/en-us/windows/win32/search/-search-sql-windowssearch-entry) and supports OleDb with connection string `provider=Search.CollatorDSO;EXTENDED PROPERTIES="Application=Windows"`.

**Comment by pmatrosov** [comment]
YDB from Yandex (https://ydb.tech/)

**Comment by parasparmar** [comment]
If possible, @MaceWindu can you look at Couchbase support for Linq2DB. 
The documents to connect to it via ADO. NET are linked here. 
[Couchbase. NET SDK & documentation](https://docs.couchbase.com/dotnet-sdk/current/hello-world/start-using-sdk.html)

**Comment by rrrevery** [comment]
There are many company and orgnization in China must use the database writen by China,such as Kingbase which is similar to postgresql,can you add support Kingbase to Linq2db

**Comment by igor-tkachev** [comment]
> There are many company and orgnization in China must use the database writen by China,such as Kingbase which is similar to postgresql,can you add support Kingbase to Linq2db

Can you provide a link to relevant documentation?

**Comment by rrrevery** [comment]
> > There are many company and orgnization in China must use the database writen by China,such as Kingbase which is similar to postgresql,can you add support Kingbase to Linq2db
> 
> Can you provide a link to relevant documentation?

Their english offical site is https://www.kingbase.com.cn/en/
Their document site is https://help.kingbase.com.cn/v8/index.html ,I didn't find english version
I konw another c# orm support Kingbase,maybe you can refer to this repo https://github.com/dotnetcore/FreeSql/tree/master/Providers/FreeSql.Provider.KingbaseES

**Comment by GirusVirus** [comment]
How do I add InterBase to the list? this db has ADO.NET support. [https://www.embarcadero.com/free-tools/ado-net-driver](url)

**Comment by to11mtm** [comment]
Not necessarily a full-on new RDBMS, but it would be interesting to add TimescaleDb extensions for the Postgres provider.

It would however likely be primarily things like extra function impls and extra create table support rather than a full-on provider however (and may be better done as a separate add-in).

**Comment by xrem** [comment]
> YDB from Yandex (https://ydb.tech/)

Regarding YDB, there is ADO.Net Data Provider available:
https://github.com/ydb-platform/ydb-dotnet-sdk/tree/main/src/Ydb.Sdk/src/Ado

@MaceWindu can you please update status in table?

**Comment by MaceWindu** [comment]
@xrem thanks for info

**Comment by dibargatin** [comment]
**[IMPLEMENTED](https://github.com/linq2db/linq2db/pull/5451)**

DuckDB
https://github.com/Giorgi/DuckDB.NET

**Comment by FreeYourMinds** [comment]
# `Caché` or now `IRIS` from Intersystems

Could you please consider adding support for `Caché` or now `IRIS` from Intersystems?

For further details refer to [Using .NET and the ADO.NET Managed Provider with Caché](https://docs.intersystems.com/latest/csp/docbook/DocBook.UI.Page.cls?KEY=GBMP)

**Comment by KirillKurdyukov** [comment]
@MaceWindu Hi!
PR for support YDB - https://github.com/linq2db/linq2db/pull/4954

**Comment by jogibear9988** [comment]
Add support for Influx 3
It supports now a SQL dialect, and has a C# provider:
https://github.com/InfluxCommunity/influxdb3-csharp/

andy they do not plan to support linq themselfes:
https://github.com/InfluxCommunity/influxdb3-csharp/issues/160

**Comment by to11mtm** [comment]
.... I think it would be helpful (and a big win for the project as far as larger adoption) if we could support Databricks;

Even ODBC for first go would be a big win since AFAIK no other ORMs or similar on .NET are as positioned as well as linq2db to implement (i.e. at least to start we can do minimal ODBC implementation and go from there.)
